sleepylyze/nrem.py

```diff
@@ -1164,56 +1164,52 @@ def calc_spin_tstats(self):
         columns = pd.MultiIndex.from_arrays([lvl1, lvl2])
         spindle_stats = pd.DataFrame(columns=columns)
         
-        #exclude non-EEG channels
-        exclude = ['EOG_L', 'EOG_R', 'EKG']
-
         # fill dataframe
-        for chan in self.spindles:
-            if chan not in exclude:
-                # calculate spindle count
-                count = len(self.spindles[chan])
-                
-                if count == 0:
-                    spindle_stats.loc[chan] = [count, None, None, None, None, None, None, None, None, None]
-                
+        for chan in self.channels:
```

Comment:
Let's set the exclude attribute at init and access self.exclude instead of self.channels within the later functions. This allows the accessed data to be more explicit and the source of bugs to be located more easily

For example, here self.spindles should be used instead of self.channels to ensure access of the the channel names from the spindle detections. They *should* be the same as the channel names from the non-excluded data, but in case they are not it's easier to find where things went wrong

This can be edited to:
```
exclude = self.exclude
for chan in self.spindles:
    if chan not in exclude:
        ...
```

Reply:
Kept exclude and channels separate in commit 1438e7f